### PR TITLE
Llama galaxy: host sampling for structured decoding 

### PR DESCRIPTION
**Pull request by sraizada-tt**
### Ticket
[Link to Github Issue](https://github.com/tenstorrent/vllm/issues/167)

This PR implements host sampling for structured decoding in the Llama 3 70B Galaxy model by adding support for returning logits instead of performing device-side sampling. This enables more flexible sampling strategies on the host side.

Key changes:

return_logits is true when sampling params is None
Modified prefill and decode forward methods to conditionally return logits based on sampling_params
Enhanced output processing to handle logits when vocab size is large

Galaxy APC: https://github.com/tenstorrent/tt-metal/actions/runs/17986421169
Galaxy demo: https://github.com/tenstorrent/tt-metal/actions/runs/17986433692

### CODE REVIEW

**Comment by sraizada-tt** [comment]
fyi - @tstescoTT

**Comment by tchedaTT** [comment]
@ pr description - I think return_logits is no longer a parameter, and now just an internal variable?

File: models/demos/llama3_70b_galaxy/tt/generator.py [hunk 1]
```diff
@@ -506,6 +540,13 @@ def read_decode_output(self, tt_out, async_read=True):
         return logits, [read_event]
 
     def process_decode_output_host(self, tt_out, is_tokens=True):
+        if isinstance(tt_out, tuple):
+            tt_out = tt_out[0]
+        if tt_out.shape[-1] >= self.model.vocab_size // 8:
```
Comment (mtairum) [inline models/demos/llama3_70b_galaxy/tt/generator.py:547]: add a comment to explain :)

**Review by mtairum — CHANGES_REQUESTED** [review]

**Review by mtairum — APPROVED** [review]
